### ISSUE TITLE
Out of memory error.

### ISSUE THREAD

**Issue by jgm1986** [mami-project/KeyServer#13]
KeyServer OutOfMemoryError when 22143 (approx) keys has been processed.

```
Exception in thread "Thread-2" java.lang.OutOfMemoryError: Java heap space
        at sun.security.ssl.InputRecord.<init>(InputRecord.java:93)
        at sun.security.ssl.HandshakeInStream.<init>(HandshakeInStream.java:63)
        at sun.security.ssl.Handshaker.activate(Handshaker.java:535)
        at sun.security.ssl.SSLEngineImpl.kickstartHandshake(SSLEngineImpl.java:729)
        at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:829)
        at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:781)
        at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624)
        at sun.net.httpserver.SSLStreams$EngineWrapper.recvAndUnwrap(SSLStreams.java:338)
        at sun.net.httpserver.SSLStreams.recvData(SSLStreams.java:406)
        at sun.net.httpserver.SSLStreams$InputStream.read(SSLStreams.java:522)
        at sun.net.httpserver.SSLStreams$InputStream.read(SSLStreams.java:591)
        at sun.net.httpserver.Request.readLine(Request.java:80)
        at sun.net.httpserver.Request.<init>(Request.java:50)
        at sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:536)
        at es.tid.keyserver.https.HttpsServerController$2.execute(HttpsServerController.java:161)
        at sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:431)
        at sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:396)
        at java.lang.Thread.run(Thread.java:745)
Exception in thread "AWT-EventQueue-0" java.lang.OutOfMemoryError: Java heap space
        at java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:57)
        at java.nio.ByteBuffer.allocate(ByteBuffer.java:335)
        at sun.nio.cs.StreamEncoder.<init>(StreamEncoder.java:195)
        at sun.nio.cs.StreamEncoder.<init>(StreamEncoder.java:175)
        at sun.nio.cs.StreamEncoder.forOutputStreamWriter(StreamEncoder.java:68)
        at java.io.OutputStreamWriter.<init>(OutputStreamWriter.java:133)
        at java.io.PrintStream.<init>(PrintStream.java:111)
        at java.io.PrintStream.<init>(PrintStream.java:175)
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:438)
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:527)
        at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
        at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:153)
        at es.tid.keyserver.core.lib.LastVersionAvailable.refreshRepoStatus(LastVersionAvailable.java:108)
        at es.tid.keyserver.core.status.KsMonitor$2.actionPerformed(KsMonitor.java:151)
        at javax.swing.Timer.fireActionPerformed(Timer.java:313)
        at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
        at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
```

**Comment by jgm1986** [comment]
![class](https://cloud.githubusercontent.com/assets/11734525/20430436/95130e2e-ad94-11e6-9c3e-2d368461138f.png)
![graphs](https://cloud.githubusercontent.com/assets/11734525/20430437/951910e4-ad94-11e6-8b7a-27094680e061.png)

**Comment by jgm1986** [comment]
Problem solved in KeyServer v0.4.0.

**Comment by jgm1986** [comment]
New profiling KeyServer v0.4.0 solving 2M of keys from 2 client connections, in parallel (1M by client).

![keyserver v0 4 0](https://cloud.githubusercontent.com/assets/11734525/20827137/34e592ba-b870-11e6-850a-c1f30b3a580f.png)


The maximum occupation of heap is 118MiB.
